### PR TITLE
Align some quantum sub-directories

### PR DESCRIPTION
**Pull request by zvecr**
## Description

* `quantum/templates/`
* `quantum/tools/`

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by fauxpark — CHANGES_REQUESTED** [review]
`eeprom_reset.hex` should be renamed to `reset.eep`, and the readme removed (the docs can cover this, it only mentions dfu-programmer, and it now applies to the whole of `util/`).

**Review by fauxpark — CHANGES_REQUESTED** [review]
Oops, the gitignore exception for the EEPROM reset should be changed too.

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -20,7 +20,6 @@
 api_data/v1
 doxygen/
 quantum/version.h
-!quantum/tools/eeprom_reset.hex
 *.bin
```
Comment (fauxpark) [inline .gitignore:23]:
```suggestion
!util/reset.eep
*.bin
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (zvecr) [reply .gitignore:23]: theres no point adding an exclude at this point

Comment (fauxpark) [reply .gitignore:23]: Fine by me.

**Review by fauxpark — APPROVED** [review]

**Pull request merged**